apm-agent-tracer/src/main/java/co/elastic/apm/agent/tracer/direct/DirectSpan.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.tracer.direct;
+
+import co.elastic.apm.agent.tracer.Span;
+
+import javax.annotation.Nullable;
+
+public interface DirectSpan<T extends DirectSpan<T>> extends AbstractDirectSpan<T>, Span<T> {
```

Comment:
I stacked these PRs as they touch quite some of the same code and I wanted to avoid lengthy merges, I should have documented that order, sorry. The `direct` package can be considered as API for creating explicit spans and recording explicit data as with open tracing and open telementry. Maybe `explicit` would be a better name for it.